pyspecdata/core.py

```diff
@@ -7255,25 +7209,26 @@ def rename(self,previous,new):
             self.fit_axis = new
         nddata.rename(self,previous,new)
         return self
-    def fit(self,set = None, set_to = None, force_analytical = False, silent = False):
+    def fit(self,set_what = None, set_to = None, force_analytical = False, silent = False):
         r'''actually run the fit'''
         if not silent: print("\n")
         if not silent: print(r'\resizebox*{!}{3in}{\begin{minipage}{\linewidth}')
-        if isinstance(set, dict):
-            set_to = list(set.values())
-            set = list(set.keys())
+        if isinstance(set_what, dict):
+            set_to = list(set_what.values())
+            set_what = list(set_what.keys())
         x = self.getaxis(self.fit_axis)
         if iscomplex(self.data.flatten()[0]):
-            if not silent: print(lsafen('Warning, taking only real part of fitting data!'))
+            if not silent: print((lsafen('Warning, taking only real part of fitting data!')))
         y = real(self.data)
         sigma = self.get_error()
         if sigma is None:
             if not silent: print('{\\bf Warning:} You have no error associated with your plot, and I want to flag this for now\n\n')
             warnings.warn('You have no error associated with your plot, and I want to flag this for now',Warning)
             sigma = ones(shape(y))
-        p_ini = real(array(self.guess())) # need the numpy format to allow boolean mask
-        if set != None:
-            self.set_indices,self.set_to,self.active_mask = self.gen_indices(set,set_to)
+        p_ini = [1.0,1.0,1.0] # hard-coded for debug
+        #p_ini = real(array(self.guess())) # need the numpy format to allow boolean mask
```

Comment:
I would just delete lines that have been commented out

Reply:
Yes, did that while addressing the previous